File: backend/internal/models/mt5.go
package models

type MT5 struct {
	MT5ID    string  `json:"mt5_id" gorm:"primaryKey;column:mt5_id"`
	UserID   string  `json:"user_id" gorm:"column:user_id"`
	Name     string  `json:"name" gorm:"column:name"`
	Token    string  `json:"token" gorm:"column:token"`
	Status   bool    `json:"status" gorm:"column:status"`
}

func (MT5) TableName() string {
	return "MT5"
}

File: frontend/types/mt5.ts
export interface addAccountProps {
	mt5_id: string;
	user_id: string;
	name: string;
	token: string;
}

export interface MT5 {
	mt5_id: string;
	user_id: string;
	name: string;
	token: string;
	status: boolean;
}

export interface OpenPosition {
	pair: string;
	type: "BUY" | "SELL";
	entry: number;
	current: number;
	lot: number;
	profit: number;
}

export interface ClosedPosition {
	pair: string;
	type: "BUY" | "SELL";
	entry: number;
	lot: number;
	profit: number;
}

export interface MT5AccountCardStats {
	equity: number;
	balance: number;
	realized_today: number;
	realized_week: number;
	is_connected: boolean;
	open_positions: OpenPosition[];
	closed_positions: ClosedPosition[];
}
